services/checks/checks.go

```diff
@@ -943,7 +881,7 @@ func (s *Service) findTargets(serviceType models.ServiceType, minPMMAgentVersion
 }
 
 // groupChecksByDB splits provided checks by database and returns three slices: for MySQL, for PostgreSQL and for MongoDB.
-func (s *Service) groupChecksByDB(checks []check.Check) (mySQLChecks, postgreSQLChecks, mongoDBChecks map[string]check.Check) {
+func (s *Service) groupChecksByDB(checks map[string]check.Check) (mySQLChecks, postgreSQLChecks, mongoDBChecks map[string]check.Check) {
```

Comment:
It requires a logger inside, and because this function is not used anywhere outside this service it's simpler to have it like a method, than pass logger as an argument.